.github/workflows/ci-cd.yml

```
@@ -19,98 +19,106 @@ jobs:
+          AIOHTTP_VERSION: ${{ matrix.aiohttp-version }}
+        run: |
+          python -c "import sys; print(f'Python version: {sys.version}')"
+          python -m pip install -U pip codecov pip-tools
+          time pip-compile requirements-dev.in
+          time pip-sync requirements-dev.txt
+      - name: Run pre-commit hooks
+        if: matrix.python-version == '3.11'
+        run: |
+          make pre-commit
+      - name: Run unittests
+        env:
+          COLOR: 'yes'
         run: |
-            echo "Predeploy step"
+          make mototest
+      - name: Upload coverage to Codecov
+        if: matrix.python-version == '3.11'
+        uses: codecov/codecov-action@v4.5.0
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+          files: ./coverage.xml
+          flags: unittests # optional
+          name: codecov-umbrella # optional
+          fail_ci_if_error: true # optional (default = false)
+          verbose: true # optional (default = false)
 
-  build-tarball:
-    name: Tarball
+
+  # based on https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python#publishing-to-pypi
+  release-build:
     runs-on: ubuntu-latest
-    needs: pre-deploy
+
+    needs:
+      - test
+
     steps:
-    - name: Checkout
-      uses: actions/checkout@v4
-      with:
-        submodules: true
-    - name: Setup Python 3.8
-      uses: actions/setup-python@v5
-      with:
-        python-version: 3.8
-    - name: Make distributions (sdist and wheel)
-      run: |
-        python -c "import sys; print(f'Python version: {sys.version}')"
-        python -m pip install -U setuptools pip wheel
-        python setup.py sdist bdist_wheel
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v4
-      with:
-        name: dist
-        path: dist
-
-  deploy:
-    name: Deploy
-    needs: [build-tarball]
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+
+      - name: Build release distributions
+        run: |
+          python -m pip install build
+          python -m build
+
+      - name: Check release distributions
+        run: |
+          python -m pip install twine
+          python -m twine check --strict dist/*
+
+      - name: Upload distributions
+        uses: actions/upload-artifact@v4
+        with:
+          name: release-dists
+          path: dist/
+
+  pypi-publish:
     runs-on: ubuntu-latest
+
+    needs:
+      - release-build
+
+    # Run only on pushing a tag
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+
     permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
       id-token: write
+
+    # Dedicated environments with protections for publishing are strongly recommended.
+    environment:
+      name: pypi
+      url: https://pypi.org/project/aiobotocore/${{github.ref_name}}
```

mustache is sparse in all the other places
```suggestion
      url: https://pypi.org/project/aiobotocore/${{ github.ref_name }}
```